File: App10/app/src/main/java/com/example/app10/MainActivity2.java
package com.example.actividad02;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    TextView View01;
    EditText Edit01;
    Button aceptarButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        View01 = findViewById(R.id.View01);

        Edit01 = findViewById(R.id.Edit01);

        aceptarButton = findViewById(R.id.aceptarButton);


        if (savedInstanceState != null) {
            String TextString = savedInstanceState.getString("textoGuardado", "");
            View01.setText(TextString);
        }

        aceptarButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String TextString = Edit01.getText().toString();
                View01.setText(TextString);
            }
        });
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("textoGuardado", View01.getText().toString());
    }
}

File: App10K/app/src/main/java/com/example/app10k/MainActivity2.kt
package com.example.app10k

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class MainActivity2 : AppCompatActivity() {
    var View01: TextView? = null
    var Edit01: EditText? = null
    var aceptarButton: Button? = null

    var BTNR2: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main2)

        View01 = findViewById(R.id.View01)

        Edit01 = findViewById(R.id.Edit01)

        aceptarButton = findViewById(R.id.aceptarButton)

        BTNR2 = findViewById(R.id.BTNR2)

        BTNR2?.setOnClickListener(View.OnClickListener { v: View? -> regresarAOriginal() })


        if (savedInstanceState != null) {
            val TextString = savedInstanceState.getString("textoGuardado", "")
            View01?.setText(TextString)
        }

        aceptarButton?.setOnClickListener(View.OnClickListener {
            val TextString = Edit01?.getText().toString()
            View01?.setText(TextString)
        })
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString("textoGuardado", View01!!.text.toString())
    }


    private fun regresarAOriginal() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}
